cloud/blockstore/libs/storage/volume/volume_actor_addclient.cpp

```diff
@@ -447,8 +448,9 @@ void TVolumeActor::ExecuteAddClient(
 
         db.WriteClient(args.Info);
 
-        if (args.Info.GetFillGeneration() > 0 &&
-                IsReadWriteMode(args.Info.GetVolumeAccessMode())) {
+        if (IsReadWriteMode(args.Info.GetVolumeAccessMode()) &&
+                (args.Info.GetFillGeneration() > 0 ||
+                State->GetMeta().GetVolumeConfig().GetIsFillFinished())) {
```

Comment:
This condition could be simplified: no need to check
(args.Info.GetFillGeneration() > 0 || State->GetMeta().GetVolumeConfig().GetIsFillFinished())) 
But we don't forbid zero fill generation when disk filling is not finished. So i decided to check this just in case.